sources/lib/api/gcp/bigquery/_udf.py

```diff
@@ -13,34 +13,27 @@
 """Google Cloud Platform library - BigQuery UDF Functionality."""
 
 import json
+import gcp.storage
 
 
 class FunctionCall(object):
```

Comment:
I think we only need to support calling UDFs within SQL.

Calling into the javascript to test locally is interesting, but I don't know how realistic, given the javascript you are writing can depend on javascript in GCS.

In fact, if you are trying to write scalar UDFs, the best way to do it right now is with the UDF is GCS as a reusable script, and the specific UDF for the specific query schema as inline. For example, I want to write a UDF for geohashing over lat+lon which I can write and store in GCS to be used by a query-specific UDF that operates over columns a,b,c,lat,lon and roundtrips the a,b,c part.

I am assuming testing with the actual BQ dremel engine is cheap enough over sample data, so why would be not always do that instead of testing in the browser... just seems like a cheap hack ... yes, I wrote it, and it was actually written as a demo over lunch at an all day offsite when someone asked me if notebooks could enable test scenarios :)